src/devices/Bh1745/Bh1745.cs

```diff
@@ -0,0 +1,430 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Drawing;
+using System.Linq;
+
+namespace Iot.Device.Bh1745
+{
+    /// <summary>
+    /// Digital color sensor Bh1745.
+    /// </summary>
+    public class Bh1745 : IDisposable
+    {
+        /// <summary>
+        /// The primary I2c address of the BH1745
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x38;
+
+        /// <summary>
+        /// The secondary I2c address of the BH1745
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x39;
+
+        /// <summary>
+        /// Gets or sets the state of the software reset.
+        /// True indicates that the initial reset is started, false that it is not started.
+        /// On reset all registers are reset and the sensor goes to power down mode.
+        /// </summary>
+        public bool SoftwareReset
+        {
+            get
+            {
+                var status = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
```

Comment:
would it make sense to cache this value once it's been read once? Presumably the only way to change it is by calling set, so after you have read once, then you will be able to know when changes to the cached value happen right?